src/libcore/sync/atomic.rs

```diff
@@ -124,6 +124,7 @@ pub fn spin_loop_hint() {
 /// [`bool`]: ../../../std/primitive.bool.html
 #[cfg(target_has_atomic = "8")]
 #[stable(feature = "rust1", since = "1.0.0")]
+#[repr(transparent)]
 pub struct AtomicBool {
```

Comment:
This is a bit iffy I think. Presumably you'd expect it to be FFI compatible with a bool, not a u8, right?

Reply:
Do we not have `AtomicBool` on platforms without `u8` atomics? I could've sworn we were using `usize`  before, which would've made the `AtomicBool` change a breaking one - assuming there is a platform where `usize` atomics exist but not byte ones.

Also, why is this not wrapping `AtomicU8`? Seems like the code would be safer/cleaner that way.

Reply:
The type has a `#[cfg(target_has_atomic = "8")]`. It should start wrapping AtomicU8, but it existed way before that type did.